src/service-implementations/lf-repo-tree-node-service.ts

```diff
@@ -125,10 +124,17 @@ export class LfRepoTreeNodeService implements LfTreeNodeService {
       const parentPath = this.getParentPath(treeNode.path);
       if (parentPath) {
         const parentEntry: FindEntryResult = await this.repoClient.entriesClient.getEntryByPath({ repoId, fullPath: parentPath });
-        if (parentEntry.entry) {
+        if (parentEntry.entry && parentEntry.entry.entryType === EntryType.Folder) {
           const foundParentEntry = parentEntry.entry;
-          foundParentEntry.fullPath = parentPath;
-          return this.createLfRepoTreeNode(foundParentEntry, repoName);
+          if (foundParentEntry.id === 1) {
+          return this.getRootTreeNodeAsync();
+          }
+          const icon: string = this.getIconForEntry(foundParentEntry.entryType);
+          const parentNode = this.createFolderNode(foundParentEntry.name, parentPath, foundParentEntry.id, foundParentEntry.entryType, icon);
```

Comment:
can icon be calculated inside createFolderNode? seems like a lot of variables being passed into createFolderNode and icon is based off something that is already being passed in

Reply:
some icons depend on the parent entry (if parent is RecordSeries, then the icon is a record). So it is hard to calculate icon inside createFolderNode unless we pass in the parent